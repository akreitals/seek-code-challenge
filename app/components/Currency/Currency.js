// @flow
import React from 'react';
import Text from '../Text/Text';

const Currency = ({
    value = 0,
    locale = 'en-AU',
    currency = 'AUD',
    ...rest
}) => (
    <Text {...rest}>
        {value
            ? value.toLocaleString(locale, { style: 'currency', currency })
            : 0}
    </Text>
);

export default Currency;
