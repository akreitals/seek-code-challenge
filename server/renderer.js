import React from 'react';
import ReactDomServer from 'react-dom/server';
import { ServerStyleSheet } from 'styled-components';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import { configureStore } from '../app/state/configureStore';
import ThemeProvider from '../app/components/ThemeProvider/ThemeProvider';

import RoutesContainer from '../app/containers/Routes/RoutesContainer';

const isProduction = process.env.NODE_ENV === 'production' || false;

const render = (html, styles) => {
    const assetsManifest =
        process.env.webpackAssets && JSON.parse(process.env.webpackAssets);

    return `
    <!DOCTYPE html>
    <html lang="en">
        <head>
            <meta charset="UTF-8">
            <meta name="viewport" content="width=device-width, initial-scale=1">
            <meta http-equiv="x-ua-compatible" content="ie=edge">
            <title>Seek Coding Challenge</title>
            ${styles}
        </head>
        <body>
            <div id="root">${html}</div>
            <script src='${
                isProduction ? assetsManifest['/vendor.js'] : 'js/vendor.js'
            }'></script>
            <script src='${
                isProduction ? assetsManifest['/app.js'] : 'js/app.js'
            }'></script>
        </body>
    </html>
    `;
};

const serverSideRender = (req, res) => {
    const context = {};
    const sheet = new ServerStyleSheet();
    const store = configureStore();

    const appWithRouter = (
        <Provider store={store}>
            <ThemeProvider>
                <StaticRouter location={req.url} context={context}>
                    <RoutesContainer />
                </StaticRouter>
            </ThemeProvider>
        </Provider>
    );

    if (context.url) {
        return res.redirect(context.url);
    }

    const html = ReactDomServer.renderToString(
        sheet.collectStyles(appWithRouter)
    );

    const styles = sheet.getStyleTags();

    return res.status(200).send(render(html, styles));
};

export default serverSideRender;
