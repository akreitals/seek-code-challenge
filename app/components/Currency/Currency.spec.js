import React from 'react';
import renderer from 'react-test-renderer';
import Currency from './Currency';

describe('Currency', () => {
    test('should render a <Currency /> component', () => {
        const json = renderer.create(<Currency value={909.91} />).toJSON();
        expect(json).toMatchSnapshot();
    });

    test('should default to the AU locale', () => {
        const json = renderer.create(<Currency value={909.91} />).toJSON();
        expect(json).toMatchSnapshot();
        expect(json.children[0]).toBe('A$909.91');
    });

    test('should allow a different current to be set', () => {
        const json = renderer
            .create(<Currency currency="EUR" value={909.91} />)
            .toJSON();
        expect(json).toMatchSnapshot();
        expect(json.children[0]).toBe('€909.91');
    });

    test('should allow a different locale to be set', () => {
        const json = renderer
            .create(<Currency locale="de-DE" currency="EUR" value={909.91} />)
            .toJSON();
        expect(json).toMatchSnapshot();
        expect(json.children[0]).toBe('€909.91');
    });
});
