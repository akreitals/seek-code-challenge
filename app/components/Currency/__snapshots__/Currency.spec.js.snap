// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency should allow a different current to be set 1`] = `
<p
  className="sc-bdVaJa hvucuj"
>
  €909.91
</p>
`;

exports[`Currency should allow a different locale to be set 1`] = `
<p
  className="sc-bdVaJa hvucuj"
>
  € 909.91
</p>
`;

exports[`Currency should default to the AU locale 1`] = `
<p
  className="sc-bdVaJa hvucuj"
>
  A$909.91
</p>
`;

exports[`Currency should render a <Currency /> component 1`] = `
<p
  className="sc-bdVaJa hvucuj"
>
  A$909.91
</p>
`;
